Sources/SwiftParser/Availability.swift

```diff
@@ -261,44 +261,46 @@ extension Parser {
   ///     platform-version → decimal-digits
   ///     platform-version → decimal-digits '.' decimal-digits
   ///     platform-version → decimal-digits '.' decimal-digits '.' decimal-digits
-  mutating func parseVersionTuple() -> RawVersionTupleSyntax {
+  mutating func parseVersionTuple(maxComponentCount: Int = 3) -> RawVersionTupleSyntax {
```

Comment:
I just changed my mind and would prefer not to default this to any value. I think every call site of this function should be forced to explicitly think about how many components it wants to support, just so it’s a conscious decision.